### ISSUE TITLE
Embed doctype, receipt date, and doc ID as PDF property metadata

### ISSUE THREAD

**Issue by orischwartz** [department-of-veterans-affairs/caseflow-efolder#319]
This is required for use in the eReader pilot, but has also been [requested in the past](https://github.com/department-of-veterans-affairs/caseflow-efolder/issues/237#issuecomment-252655125).

Currently we include the following attributes as part of [file naming scheme](https://github.com/department-of-veterans-affairs/caseflow-efolder/blob/master/app/models/document.rb#L18):
* doctype
* receipt date
* doc ID

Certain 3rd party tools can more easily process this data when it's part of the [PDF's metadata](https://helpx.adobe.com/acrobat/using/pdf-properties-metadata.html).

#### Implementation options:
* Use a ruby library that wraps pdftk, which we can easily install (e.g., https://github.com/blambeau/pdf-toolkit/)
* Invoke pdftk directly via [CLI `update_info`](https://www.pdflabs.com/docs/pdftk-man-page/)

#### Risks:
* Modifies original document, could introduce PDF incompatibilities
* More processing/IO intensive

**Comment by orischwartz** [comment]
Able to generate the following metadata:

```
InfoBegin
InfoKey: Receipt Date
InfoValue: 2013-01-01

InfoBegin
InfoKey: Document Type
InfoValue: Form 9

InfoBegin
InfoKey: CreationDate
InfoValue: D:20131129152618-05'00'

InfoBegin
InfoKey: Document Id
InfoValue: 4f6180f6-c16f-11e6-a4a6-cec0c932ce01
```

```
$ pdftk in.pdf update_info info.txt output out.pdf
```
![image](https://cloud.githubusercontent.com/assets/16327036/21157080/31e72c0c-c146-11e6-8973-41dd5ba0d518.png)

**Comment by orischwartz** [comment]
@amprokop if not complete by week of 12/26 just assign back to me and I'll take a stab at it for week of 1/2

**Comment by orischwartz** [comment]
Changes required:

* Add  `- sudo apt-get install pdftk` to the `before_script` section of `.travis.yml`.
* Add dependency for [SafeShell](https://github.com/envato/safe_shell)
* Modify [`save_document_file`](https://github.com/department-of-veterans-affairs/caseflow-efolder/blob/master/app/services/download_documents.rb#L69) to do the following when input file is a PDF:
  1. Define a hash on attributes to be written
  1. Write attribute hash in the PDF attribute format to a temporary file (can be read from stdin but we'll read the PDF from stdin; not sure if there's a way to read both and avoid disk).
  1. Run pdftk, reading the pdf contents from stdin and writing to specified file name
* Modify deployment process to install `pdftk`

**Comment by orischwartz** [comment]
Implemented in #344
